test/unit/runner.spec.js

```diff
@@ -487,6 +487,32 @@ describe('Runner', function() {
         });
         runner.failHook(hook, err);
       });
+
+      it('should not hang if the error message is ridiculously long', function(done) {
+        var hook = new Hook();
+        var message = [];
+        // mock a long message
+        for (var i = 0; i < 20000; i++) {
+          var line = stack[i % stack.length];
+          message[i] = line[Math.floor(Math.random() * line.length)];
+        }
+        var err = new Error(message);
+        // Fake stack-trace
```

Comment:
Error.message or Error.stack, are both causing problems when too large?

Reply:
Think he's saying `Error.message` is causing the problem, but it's the first line of the stacktrace...

Reply:
this is not necessarily the case:

```js
throw new Error(`foo
bar
baz
`);
```

Reply:
@juergba `err.stack` is composed of `err.message` and the real stack. Skipping the first line is my first thought but `err.message` could be multiline. My second try is filtering the real stack by checking `/^\s+at ` first but the test cases already constructed (in the file below) is far complicated than I thought. 

Removing `/^\(?.+/` is the ideal fix AFAICT because it fixes the problem and won't introduce regression.